include/RAJA/policy/cuda/policy.hpp

```diff
@@ -159,6 +159,30 @@ struct AvoidDeviceMaxThreadOccupancyConcretizer
   }
 };
 
+
+enum struct reduce_algorithm : int
+{
+  combine_last_block,
+  init_device_combine_atomic_block,
+  init_host_combine_atomic_block
+};
+
+enum struct block_communication_mode : int
+{
+  device_fence,
+  block_fence
+};
+
+template < reduce_algorithm t_algorithm, block_communication_mode t_comm_mode,
+           size_t t_replication, size_t t_atomic_stride >
+struct ReduceTuning
+{
+  static constexpr reduce_algorithm algorithm = t_algorithm;
+  static constexpr block_communication_mode comm_mode = t_comm_mode;
+  static constexpr size_t replication = t_replication;
+  static constexpr size_t atomic_stride = t_atomic_stride;
+};
```

Comment:
Here are more types that the user is not really expected to use, but can use to fine tune things.

Reply:
Add example usage of this to cookbook?